### ISSUE TITLE
Adding more reagents makes GuideEntryPrototypeTests fail, as it then hits UI update limit

### ISSUE THREAD

**Issue by TheArturZh** [space-wizards/space-station-14#24380]
## Description

https://github.com/space-wizards/space-station-14/pull/23987 made chem guidebook grow the number of its UI elements quite significantly. If a bit more reagents and reagent sources are added, it will make GuideEntryPrototypeTests.ValidatePrototypeContents test fail, as it will hit style update limit and throw a warning.
We have already ran into this issue on downstream.

RobustToolbox/Robust.Client/UserInterface/UserInterfaceManager.cs:236
```
CLIENT: 7,117s [WARN] ui: Hit style update limit. Queued: 397. Next in queue:  (Collapsible). Parent: RecipesContainer (BoxContainer)
```

Potential solution is to generate UI elements only when source/recipe Collapsable is expanded.

**Reproduction**

Add a bunch more reagents/reagent sources (duplicating everything in Prototypes/Reagents/chemicals.yml was enough), run the GuideEntryPrototypeTests.ValidatePrototypeContents test

**Comment by metalgearsloth** [comment]
The update limit is 25,000 so the fact this is even being hit is extremely concerning.

**Comment by TheArturZh** [comment]
fixed in #24427
